osu.Game.Rulesets.Osu/Skinning/SmokeSegment.cs

```diff
@@ -234,9 +236,14 @@ public sealed override void Draw(IRenderer renderer)
                 if (points.Count == 0)
                     return;
 
-                rotationIndex = 0;
+                quadBatch ??= renderer.CreateQuadBatch<TexturedVertex2D>(200, 4);
+
+                if (points.Count > quadBatch.Size * 4 && quadBatch.Size != 10922)
+                {
+                    int batchSize = Math.Min((int)(quadBatch.Size * 1.5f), 10922);
+                    quadBatch = renderer.CreateQuadBatch<TexturedVertex2D>(batchSize, 4);
+                }
```

Comment:
- `VertexBatch<T>.Size` actually indicates the number of elements (in this case, quads), contrary to the documentation which states number of vertices. So if each point is adding 4 vertices (= `IRenderer.VERTICES_PER_QUAD`) then I'd expect the first part of the condition to be `if (points.Count > quadBatch.Size)`.
- `10922` can be replaced with `IRenderer.MAX_QUADS`.
- Any reason for the 1.5 expansion factor? List implementations typically [scale with a factor of 2](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs,456) which sounds fine here.

Reply:
I think some C++ `std::vector` implementations use 1.5x, but 2x seems equally common based on a quick search. If .NET `Lists` use 2x then we can go with 2x, I didn't really have any particular reason for 1.5x. (I see some references to 1.5x being closer to the golden ratio, but I honestly have no idea how that's relevant)